src/utils/job_params_util.py

```diff
@@ -61,3 +79,276 @@ def get_resource_params_from_job_params(params):
+            "sku",
+            "gpu_limit",
+            "cpu_request",
+            "cpu_limit",
+            "memory_request",
+            "memory_limit",
+        ]
+        for key in checklist:
+            if self.__dict__[key] is None:
+                return False
+        return True
+
+    def gen_sku(self):
+        sku = self.params.get("sku", "")
+        if sku != "":
+            self.sku = sku
+            return
+
+        sku_list = self.get_sku_list()
+        if len(sku_list) > 0:
+            self.sku = sku_list[0]
+
+    def gen_gpu(self):
+        # TODO: Deprecate resource_gpu in the future
+        resource_gpu = self.params.get("resourcegpu")
+        self.gpu_limit = self.params.get("gpu_limit", resource_gpu)
+
+        gpu_limit = self.get_gpu_limit()
+
+        if self.gpu_limit is None:
+            self.gpu_limit = gpu_limit
+        if self.gpu_limit is not None:
+            self.gpu_limit = int(self.gpu_limit)
+
+    def gen_cpu(self):
+        self.cpu_request = self.params.get("cpurequest")
+        self.cpu_limit = self.params.get("cpulimit", self.cpu_request)
+
+        request, limit = self.get_cpu_request_and_limit()
+
+        if self.cpu_request is None:
+            self.cpu_request = request
+        if self.cpu_limit is None:
+            self.cpu_limit = limit
+
+    def gen_memory(self):
+        self.memory_request = self.params.get("memoryrequest")
+        self.memory_limit = self.params.get("memorylimit", self.memory_request)
+
+        request, limit = self.get_memory_request_and_limit()
+
+        if self.memory_request is None:
+            self.memory_request = request
+        if self.memory_limit is None:
+            self.memory_limit = limit
+
+    def get_sku_list(self):
+        gpu_sku_list = [
+            sku_val for sku_val, _ in self.quota.get("gpu", {}).items()
+        ]
+        cpu_sku_list = [
+            sku_val for sku_val, _ in self.quota.get("cpu", {}).items()
+            if self.metadata.get("gpu", {}).get(sku_val) is None
+        ]
+        if len(cpu_sku_list) == 0:
+            cpu_sku_list = gpu_sku_list
+
+        gpu_limit = get_gpu_limit(self.params)
+        if gpu_limit > 0:
+            return gpu_sku_list
+        else:
+            return cpu_sku_list
+
+    @override
+    def get_gpu_limit(self):
+        return get_gpu_limit(self.params)
+
+    @override
+    def get_cpu_request_and_limit(self):
+        if self.gpu_limit > 0:
+            # For a gpu job, proportionally assign cpu according to gpu.
+            request, limit = self.get_cpu_proportional()
+        else:
+            # For a cpu job, assign system default.
+            request, limit = self.get_default_cpu_request_and_limit()
+        return request, limit
+
+    @override
+    def get_memory_request_and_limit(self):
+        if self.gpu_limit > 0:
+            # For a gpu job, proportionally assign memory according to gpu.
+            request, limit = self.get_memory_proportional()
```

Comment:
I still have some concerns here. User do not know how much memory they can get if they didn't specify sku and memory request/limit. Maybe we need to expose environment variables about the resource. I'm not sure how users thinking about this.

Reply:
Let's discuss and resolve this concern in a future PR.